InpaintHTTP/MainMod.cs

```diff
@@ -22,84 +22,88 @@ public MainMod()
-                var donors = new List<ZsImage>();
-                foreach (var file in this.Request.Files)
+                try
                 {
-                    Bitmap tempBitmap;
-                    byte[] ByteImg = new byte[file.Value.Length];
-                    file.Value.Read(ByteImg, 0, (int)file.Value.Length);
-                    using (MemoryStream ms = new MemoryStream(ByteImg))
-                        tempBitmap = new Bitmap(ms);
-
-                    if (BitmapImg == null)
+                    Console.WriteLine("[" + DateTime.Now.ToString("HH:mm:ss.fffff") + "] Incomming request from " + this.Request.UserHostAddress);
+                    if (this.Request.Files.Count() < 2)
                     {
-                        BitmapImg = tempBitmap;
+                        Console.WriteLine("[" + DateTime.Now.ToString("HH:mm:ss.fffff") + $"] Error, {this.Request.Files.Count()} files found");
+                        return "Err";
                     }
-                    else if (BitmapMask == null)
+
+                    Bitmap BitmapImg = null, BitmapMask = null;
+                    var donors = new List<ZsImage>();
+                    foreach (var file in this.Request.Files)
                     {
-                        BitmapMask = tempBitmap;
-                    }
-                    else {
-                        donors.Add(tempBitmap.ToArgbImage());
+                        Bitmap tempBitmap;
+                        byte[] ByteImg = new byte[file.Value.Length];
+                        file.Value.Read(ByteImg, 0, (int)file.Value.Length);
+                        using (MemoryStream ms = new MemoryStream(ByteImg))
+                            tempBitmap = new Bitmap(ms);
+
+                        if (BitmapImg == null)
+                        {
+                            BitmapImg = tempBitmap;
+                        }
+                        else if (BitmapMask == null)
+                        {
+                            BitmapMask = tempBitmap;
+                        }
+                        else
+                        {
+                            donors.Add(tempBitmap.ToArgbImage());
+                        }
                     }
-                }
 
-                var imageArgb = ConvertToArgbImage(BitmapImg);
-                var markupArgb = ConvertToArgbImage(BitmapMask);
+                    var imageArgb = ConvertToArgbImage(BitmapImg);
+                    var markupArgb = ConvertToArgbImage(BitmapMask);
 
-                var inpainter = new Inpainter();
-                var settings = new InpaintSettings
-                {
-                    MaxInpaintIterations = 15,
-                    PatchDistanceCalculator = ImagePatchDistance.Cie76,
-                    PatchSize = 11
-                };
-                if (!Int32.TryParse(Environment.GetEnvironmentVariable("MAX_INPAINT_ITERATIONS"), out settings.MaxInpaintIterations))
```

Comment:
This environment variables are good to have to be able to configure default settings when run within Docker container.
It is better to leave them and override their values if new values are provided in the request.

Reply:
Now that there's the API, I think the environment variable should take on a different purpose. API values for what the user wants, environment variables to specify the host's acceptable value range. The max inpaint iterations is a good example of this: the host/system administrators will want to limit this value to prevent an extremely large value from consuming all the server's resources, and the user will want to specify a larger or smaller value (within the server's allowable range) depending on how long they want to wait and the quality they are going for. So the pseudocode to determine the max inpaint iterations for a request should be (imo):
```
Math.min(api_value || default_value, environment_variable_max_value || default_max_value)
```
For the patch distance environment variable you would specify which distance algorithms you would allow, and in the patch size you would set the minimum patch size (because a smaller patch size means more work for the server). If a user makes a request outside of the limited ranges, then the values should either be clamped or an error is returned. An API endpoint to get the server settings might also be helpful.

Reply:
In general I like this suggestion! 👍 

The assumption about the relation between the patch size and load for server is not correct -> bigger patch causes more work to compare patches.

Reply:
Sounds good!

TBH I think having the constraints set in the env variables should be enough and the `default max` sounds redundant to me. 🤔 

The settings are deserialized from the request(if provided) otherwise a default are used. Then the constrains provided in the env variables(if any) applied to the settings.

Reply:
> The settings are deserialized from the request(if provided) otherwise a default are used. Then the constrains provided in the env variables(if any) applied to the settings.
Alright, I have created a ``defaultSettings`` that gets created at startup, and then every request a ``userSettings`` will get created and it checks if ``userSettings.PatchSize`` and ``user.Settings.MaxInpaintIterations`` are greater then the ones set in ``defaultSettings``. I am not 100% sure but I guess I'm pretty sure to what ya want.

Reply:
> If a user makes a request outside of the limited ranges, then the values should either be clamped or an error is returned. An API endpoint to get the server settings might also be helpful.

@scribblemaniac Something like that is good to have, I have chosen to no block the request and let it proceed by lowering the values. Of course, we should let the user know that their selected values are exceeding the server maximum. Therefore I have added the `/api/settings` API so we can implement a client-side feature that visually shows the user the maximum value, or warns the user when it exceeds a maximum value.